File: Python/PrimalNumberKeeper.py
#Класс памяти, который содержит хранителя чисел и методы для него.
class Memory():
        
        #Список для хранения простых чисел.
        _keeper=[]
        
        #Заполнить хранителя(список для хранения) первыми значениями.
        @classmethod
        def CreateKeeper(self):
                self._keeper = [ 3, 5, 7 ]

        
        #Добавить к концу хранителя список чисел.
        @classmethod
        def AddKeeper(self, addList):
            for i in addList:
                self._keeper.append(i)

        #Получить хранителя(список найдены протых чисел).
        @classmethod
        def GetKeeper(self):
                return self._keeper

        #Получить хранителя(список найдены протых чисел).
        @classmethod
        def SetKeeper(self, newKeeper):
                if len(newKeeper)>len(self._keeper):
                        self._keeper=newKeeper

File: CSharp/PrimalNumbers/PrimalNumberKeeper.cs
using System.Collections.Generic;

namespace PrimalNumbers
{
    class Memory
    {
        private List<ulong> keeper;
        public Memory()
        {
            CreateKeeper();
        }

        private void CreateKeeper()
        {
            keeper = new List<ulong> { 3, 5, 7 };
        }
        
        public void AddKeeper(List<ulong> addList)
        {
            keeper.AddRange(addList);
        }

        public List<ulong> GetKeeper()
        {
            return keeper;
        }

        public void SetKeeper(List<ulong> newKeeper)
        {
            if (keeper == newKeeper)
                return;
            if (newKeeper.Count > keeper.Count)
                keeper = newKeeper;
        }
    }
}
